src/main/java/net/rubygrapefruit/platform/file/FileWatcherCallback.java

```diff
@@ -0,0 +1,10 @@
+package net.rubygrapefruit.platform.file;
+
+/**
+ * A callback that is invoked whenever a path has changed.
+ */
+public interface FileWatcherCallback {
+    // Invoked from native code
+    @SuppressWarnings("unused")
+    void pathChanged(String path);
```

Comment:
It would be good to clarify the contract here a bit more:

- Are these absolute paths?
- What happens when something is renamed or moved? deleted? created?
- Does the callback implementation need to care about thread safety?

Also, what happens when the callback throws an exception?

Reply:
Added docs for all this. Callback exceptions are currently ignored, need to fix that eventually.